src/components/Lottie/index.tsx

```diff
@@ -30,11 +33,40 @@ function Lottie({source, webStyle, ...props}: Props, ref: ForwardedRef<LottieVie
 
     const aspectRatioStyle = styles.aspectRatioLottie(source);
 
-    // If the image fails to load, app is in background state, animation file isn't ready, or the splash screen isn't hidden yet,
+    const browser = Browser.getBrowser();
+    const [hasNavigatedAway, setHasNavigatedAway] = React.useState(false);
+    const navigationContainerRef = useContext(NavigationContainerRefContext);
+    const navigator = useContext(NavigationContext);
+
+    useEffect(() => {
+        if (!browser || !navigationContainerRef || !navigator) {
+            return;
+        }
+        const unsubscribeNavigationFocus = navigator.addListener('focus', () => {
+            setHasNavigatedAway(false);
+        });
+        return unsubscribeNavigationFocus;
+    }, [browser, navigationContainerRef, navigator]);
+
+    useEffect(() => {
+        if (!browser || !navigationContainerRef || !navigator) {
+            return;
+        }
+        const unsubscribeNavigationBlur = navigator.addListener('blur', () => {
+            const state = navigationContainerRef.getRootState();
+            const targetRouteName = state?.routes?.[state?.index ?? 0]?.name;
+            if (targetRouteName !== NAVIGATORS.RIGHT_MODAL_NAVIGATOR) {
```

Comment:
```suggestion
            if (![NAVIGATORS.RIGHT_MODAL_NAVIGATOR, NAVIGATORS.LEFT_MODAL_NAVIGATOR].includes(targetRouteName)) {
```

Reply:
That will be better. Thanks for pointing it out!